File: src/main/java/DahRealPanda/plugins/untitled1/colony/model/ItemRef.java
package DahRealPanda.plugins.untitled1.colony.model;

/**
 * A reference to an item/block used as a texture key.
 *
 * <p>The {@code itemKey} is the registry name ({@code namespace:path}) optionally suffixed
 * with {@code #<8charHash>} when NBT is relevant (e.g. Domum Ornamentum textured blocks) so
 * that distinct variants map to distinct texture PNGs.</p>
 */
public class ItemRef {
    public String registryName;
    public String nbtHash; // optional, null when not NBT-relevant

    public ItemRef() {
    }

    public ItemRef(String registryName, String nbtHash) {
        this.registryName = registryName;
        this.nbtHash = nbtHash;
    }

    /** Full texture key: {@code namespace:path} plus {@code #hash} when present. */
    public String key() {
        if (nbtHash == null || nbtHash.isEmpty()) {
            return registryName;
        }
        return registryName + "#" + nbtHash;
    }
}

File: common/src/main/kotlin/DahRealPanda/plugins/colonyweb/model/ItemRef.kt
package DahRealPanda.plugins.colonyweb.model

/**
 * A reference to an item/block used as a texture key.
 *
 * The [itemKey] is the registry name (`namespace:path`) optionally suffixed
 * with `#<8charHash>` when NBT is relevant (e.g. Domum Ornamentum textured blocks) so
 * that distinct variants map to distinct texture PNGs.
 */
data class ItemRef(
    @JvmField var registryName: String = "",
    @JvmField var nbtHash: String? = null // optional, null when not NBT-relevant
) {
    /** Full texture key: `namespace:path` plus `#hash` when present. */
    fun key(): String {
        if (nbtHash.isNullOrEmpty()) {
            return registryName
        }
        return "$registryName#$nbtHash"
    }

    companion object {
        private const val serialVersionUID = 1L
    }
}
